src/NuGet.Clients/NuGet.CommandLine/Common/MSBuildProjectSystem.cs

```diff
@@ -46,17 +48,71 @@ public NuGetFramework TargetFramework
         {
             get
             {
-                string moniker = GetPropertyValue("TargetFrameworkMoniker");
+                string moniker = GetTargetFrameworkString();
+
                 if (String.IsNullOrEmpty(moniker))
                 {
                     return null;
                 }
-                return NuGetFramework.Parse(moniker);
+
+                var framework = NuGetFramework.Parse(moniker);
+
+                // if the framework is .net core 4.5.1 return windows 8.1
+                if (framework.Framework.Equals(FrameworkConstants.FrameworkIdentifiers.NetCore)
+                    && framework.Version.Equals(Version.Parse("4.5.1.0")))
+                {
+                    return new NuGetFramework(FrameworkConstants.FrameworkIdentifiers.Windows, Version.Parse("8.1"), framework.Profile);
+                }
+                // if the framework is .net core 4.5 return 8.0
+                if (framework.Framework.Equals(FrameworkConstants.FrameworkIdentifiers.NetCore)
+                    && framework.Version.Equals(Version.Parse("4.5.0.0")))
+                {
+                    return new NuGetFramework(FrameworkConstants.FrameworkIdentifiers.Windows, Version.Parse("8.0"), framework.Profile);
+                }
+
+                return framework;
             }
         }
 
         private dynamic Project { get; }
 
+        private string GetTargetFrameworkString()
+        {
+            var extension = GetPropertyValue("ProjectExt");
+
+            // Check for JS project
+            if (StringComparer.OrdinalIgnoreCase.Equals(JSProjectExt, extension))
+            {
+                // JavaScript apps do not have a TargetFrameworkMoniker property set.
+                // We read the TargetPlatformIdentifier and TargetPlatformVersion instead
+                var platformIdentifier = GetPropertyValue("TargetPlatformIdentifier");
+                var platformVersion = GetPropertyValue("TargetPlatformVersion");
+
+                // use the default values for JS if they were not given
+                if (string.IsNullOrEmpty(platformVersion))
+                {
+                    platformVersion = "0.0";
+                }
+
+                if (string.IsNullOrEmpty(platformIdentifier))
+                {
+                    platformIdentifier = "Windows";
+                }
+
+                return string.Format(CultureInfo.InvariantCulture, "{0}, Version={1}", platformIdentifier, platformVersion);
+            }
+
+            // Check for C++ project
+            if (StringComparer.OrdinalIgnoreCase.Equals(VCXProjextExt, extension))
+            {
+                // The C++ project does not have a TargetFrameworkMoniker property set. 
+                // We hard-code the return value to Native.
+                return "Native, Version=0.0";
```

Comment:
It's as part of EnvDTEProjectUtility which is part of PackageManagement.VisualStudio project... I didn't want to create commandline dependency on this project neither i found a common place to keep this logic at one place.

More over figuring out project type is different at both the places so thought of keeping separate, but will love to get a suggestion to have a common code for this.